wa/workloads/lmbench/__init__.py

```diff
@@ -0,0 +1,170 @@
+        Parameter('size', kind=list_or_string, default="4m",
+                  description='Data set size for lat_mem_rd bw_mem tests.'),
+        Parameter('mem_category', kind=list_or_string,
+                  default=('rd', 'wr', 'cp', 'frd', 'fwr', 'fcp', 'bzero', 'bcopy'),
+                  description='List of memory catetories for bw_mem test.'),
+        Parameter('parallelism', kind=int, default=None,
+                  description='Parallelism flag for tests that accept it.'),
+        Parameter('warmup', kind=int, default=None,
+                  description='Warmup flag for tests that accept it.'),
+        Parameter('repetitions', kind=int, default=None,
+                  description='Repetitions flag for tests that accept it.'),
+        Parameter('force_abi', kind=str, default=None,
+                  description='''
+                  Override device abi with this value. Can be used to force
+                  arm32 on 64-bit devices.
+                  '''),
+        Parameter('run_timeout', kind=int, default=900,
+                  description="""
+                  Timeout for execution of the test.
+                  """),
+        Parameter('loops', kind=int, default=1, constraint=lambda x: x > 0,
+                  description="""
+                  Specifies the number of times the benchmark will be run in a
+                  "tight loop", i.e. without performaing setup/teardown
+                  inbetween. This parameter is distinct from "repetitions", as
+                  the latter takes place within the benchmark and produces a
+                  single result.
+                  """),
+        Parameter('cpus', kind=cpu_mask, default=0, aliases=['taskset_mask'],
+                  description="""
+                  Specifies the CPU mask the benchmark process will be pinned to.
+                  """),
+    ]
+
+    def setup(self, context):
+
+        abi = self.target.abi
+        if self.force_abi:
+            abi = self.force_abi
+
+        # self.test has been pre-validated, so this _should_ only fail if there's an abi mismatch
+        host_exe = context.resolver.get(Executable(self, abi, self.test))
+        self.target_exe = self.target.install(host_exe)
+        self.commands = []
+
+        setup_test = getattr(self, '_setup_{}'.format(self.test))
+        setup_test()
+
+        for _ in range(self.times):
+            for command in self.commands:
+                self.target.execute(command, timeout=self.run_timeout)
+
+    def update_result(self, context):
+        host_file = os.path.join(context.output_directory, 'lmbench.output')
+        target_file = self.target.path.join(self.target.working_directory, 'lmbench.output')
+        self.target.pull_file(target_file, host_file)
+        context.add_artifact('lmbench', host_file, 'data')
+
+    def extract_results(self, context):
+        results_path = self.target.path.join(self.target.external_storage, "devlib-target", "lmbench.output")
```

Comment:
Avoid using the hard-coded path `devlib-target`, this should probably be `self.target.path.join(self.target.working_directory, 'lmbench.output')` instead.

Reply:
or, more concisely, `self.target.get_workpath('lmbench.output')`